lib/api/v3/queries/queries_api.rb

```diff
@@ -41,14 +41,27 @@ class QueriesAPI < ::API::OpenProjectAPI
             before do
               @query = Query.find(params[:id])
               @representer = QueryRepresenter.new(@query)
+              authorize_as_not_found(:show)
```

Comment:
Was wondering what that would actually mean, I don't find the naming to be precise. Not saying that I could come up with the best suggestion, but was thinking about `authorize_or_not_found`.